### PR TITLE
wip elemental equilibrium

### PR DESCRIPTION
**Pull request by wfarr**

### CODE REVIEW

**Comment by wfarr** [comment]
Testing on Beta indicates that _some_ spells (such as Lava Burst and Elemental Blast) are both generating their intended buff, as well as the buff they'd normally consume while others (Lightning Bolt) are not, so putting a pin in this for a future build.

But enjoy the 20% buff to fire damage, the 25% buff to nature damage, and the 30% buff to frost damage for random spells while it lasts. Elemental is finally viable? 🦜

File: engine/class_modules/sc_shaman.cpp [hunk 1]
```diff
@@ -1485,12 +1530,73 @@ struct shaman_spell_t : public shaman_spell_base_t<spell_t>
...
+        triggers_elemental_equilibrium_fire = true;
+      }
+
+      if ( dbc::is_school( this->school, SCHOOL_FIRE ) )
+      {
+        triggers_elemental_equilibrium_frost = true;
+      }
+
+      // only harmful spells can proc/consume the buff
+      if ( execute_state->action->harmful )
```
Comment (wfarr) [inline engine/class_modules/sc_shaman.cpp:1559]: We can remove this guard. Turns out the following non-harmful spells proc it too:

* Healing Stream Totem
* Tremor Totem
* Capacitor Totem
* Fire Elemental
* Earth Elemental
* Earthbind Totem
* Purge
* Wind Shear
* Chain Heal
* Healing Surge
* Spiritwalker's Grace
* Cleanse Spirit
* Flametongue Weapon
* Far Sight
* Water Walking
* Liquid Magma Totem
* Wind Rush Totem
* probably more

The following spells I couldn't test:
* Hex

The following did NOT proc anything:
* Astral Shift
* Lightning Shield
* Stormkeeper
* Vesper Totem
* Summon Steward

**Comment by scamille** [comment]
Do you have plans to revive/continue this before it gets too stale or impossible to merge?

**Comment by wfarr** [comment]
> Do you have plans to revive/continue this before it gets too stale or impossible to merge?

Not really.

The legendary is so incredibly bugged and/or inconsistent with its own tooltips on beta, I can't imagine it goes live in its current state, so my plan was simply to wait for Blizzard to fix it before I bothered putting a bunch of additional time/effort into replicating its current behaviors.

Happy to close and implement this from scratch at a later date.

**Comment by scamille** [comment]
Oh I did not read the full commit message of your second commit, where you already explained that.
My comment was in no way intended to get this closed, I just thought it might make sense to ask since there was quite a bit of work going on with shamans shortly before pre-patch hit and the branch was already in conflict.
Anyway, thanks for the info. Once the legendary has been finalized you can then either rebase the implementation you have here or really start from scratch if Blizzard does change it in a major way. We'll see.

**Pull request closed without merging**